### PR TITLE
sync_approved_pr: Speeding up execution

### PR DESCRIPTION
**Pull request by laukik-hase**
- Cloning base ESP-IDF from GitHub rather than GitLab
- Cloning only the required branches from remote (master and the necessary PR branch)
- Added PR URL in MR description

### CODE REVIEW

File: github_pr_to_internal_pr/github_pr_to_internal_pr.py [hunk 1]
```diff
@@ -62,10 +52,21 @@ def setup_project(project_fullname):
...
+
+    git = Git(project_name)
+    GITLAB_REMOTE = 'gitlab'
+
+    print('Adding and fetching the internal remote...')
+    git.remote('add', GITLAB_REMOTE, gl_project_url)
+    git.pull(GITLAB_REMOTE, 'master')
+
+    print('Updating submodules...')
+    git.submodule('update', '--init', '--recursive')
```
Comment (igrr) [inline github_pr_to_internal_pr/github_pr_to_internal_pr.py:68]: Question, is it really necessary to initialize submodule contents? I think merging or rebasing will work even without this step.

File: github_pr_to_internal_pr/github_pr_to_internal_pr.py [hunk 2]
```diff
@@ -253,7 +237,8 @@ def main():
```
Comment (igrr) [inline github_pr_to_internal_pr/github_pr_to_internal_pr.py:237]: Is it possible to set the MR title to `pr_title_desc + ' (Github PR)'` to make such MRs easier to distinguish? (one line above)

File: github_pr_to_internal_pr/github_pr_to_internal_pr.py (hunk 1 above)
Comment (mahavirj) [reply github_pr_to_internal_pr/github_pr_to_internal_pr.py]: @igrr What if PR has modification to submodule itself, like updating third library pointer?

**Review by dobairoland — APPROVED** [review]
Thanks! LGTM. I have one question which can be done later as an extension of the possibilities.

File: github_pr_to_internal_pr/github_pr_to_internal_pr.py (hunk 1 above)
Comment (laukik-hase) [reply github_pr_to_internal_pr/github_pr_to_internal_pr.py]: I have tested the action locally (submodule as changes in PR) and it works as expected. Resolving the comment.

File: github_pr_to_internal_pr/github_pr_to_internal_pr.py [hunk 3]
```diff
@@ -62,10 +52,18 @@ def setup_project(project_fullname):
...
     print('Cloning repository...')
-    Git(".").clone(gl_project_url, recursive=True)
+    Git(".").clone(project_html_url, branch='master', single_branch=True, recursive=True)
+
+    git = Git(project_name)
+    GITLAB_REMOTE = 'gitlab'
+
+    print('Adding and fetching the internal remote...')
+    git.remote('add', GITLAB_REMOTE, gl_project_url)
+    git.pull(GITLAB_REMOTE, 'master')
```
[the remark this answers is not in the document]
Comment (laukik-hase) [reply github_pr_to_internal_pr/github_pr_to_internal_pr.py:65]: Hello, @dobairoland 
Yes, we can implement this feature in the next stage.

**Review by mahavirj — APPROVED** [review]

**Pull request merged**